server/src/main/java/org/elasticsearch/index/mapper/ParsedDocument.java

```diff
@@ -171,4 +178,9 @@ public String toString() {
     public String documentDescription() {
         return "id";
     }
+
+    public DocumentSizeObserver getDocumentSizeObserver() {
```

Comment:
QQ: remind me why do we need this here?

Reply:
ParsedDocument is serving as a 'wrapper'/carrier for the DocumentSizeObserver that has to be used in IndexEngine.
I could change this to be a 'long' but with that design we hide that detail, and only the reporting code in `onIndexingCompleted(ParsedDocument) is calling `parsedDocument.getDocumentSizeObserver.normalizedBytesParsed`